### PR TITLE
10189 story

### PR DESCRIPTION
**Pull request by ujahio**
Script to run in upper environments: `scripts/dynamo/setup-clerk-of-court-config.sh`

### CODE REVIEW

File: shared/src/business/useCases/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlInteractor.ts [hunk 1]
```diff
@@ -49,16 +49,33 @@ export const createCourtIssuedOrderPdfFromHtmlInteractor = async (
   const { caseCaptionExtension, caseTitle } = getCaseCaptionMeta(caseDetail);
   const { docketNumberWithSuffix } = caseDetail;
 
+  const isNoticeEvent = eventCode === 'NOT'; // todo: use a constant
```
Comment (ujahio) [inline shared/src/business/useCases/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlInteractor.ts]: resolve comment.

File: shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.test.ts [hunk 2]
```diff
@@ -26,6 +26,12 @@ describe('completeDocketEntryQCInteractor', () => {
     applicationContext
       .getPersistenceGateway()
       .getLock.mockImplementation(() => mockLock);
+    applicationContext
```
Comment (ujahio) [inline shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.test.ts:29]: update test to account for the call with 'nameOfClerk' and 'titleOfClerk'

File: shared/src/business/useCases/trialSessions/generateNoticeOfChangeOfTrialJudgeInteractor.ts [hunk 3]
```diff
@@ -4,15 +4,16 @@ import { TRIAL_SESSION_SCOPE_TYPES } from '../../entities/EntityConstants';
...
-/**
- * generateNoticeOfChangeOfTrialJudgeInteractor
- *
- * @param {object} applicationContext the application context
- * @param {object} providers the providers object
- * @param {string} providers.docketNumber the docketNumber for the case
- * @param {string} providers.trialSessionInformation the trial session information
- * @returns {Uint8Array} notice of trial session pdf
- */
+// check if another type for FormattedTrial exists
```
Comment (ujahio) [inline shared/src/business/useCases/trialSessions/generateNoticeOfChangeOfTrialJudgeInteractor.ts]: check if we can use an already created Trial Info type

File: shared/src/business/useCases/trialSessions/generateNoticeOfChangeToRemoteProceedingInteractor.test.ts [hunk 4]
```diff
@@ -33,6 +33,13 @@ describe('generateNoticeOfChangeToRemoteProceedingInteractor', () => {
         judge: { name: 'Test Judge' },
       }));
 
+    applicationContext
```
Comment (ujahio) [inline shared/src/business/useCases/trialSessions/generateNoticeOfChangeToRemoteProceedingInteractor.test.ts:36]: Update assertions of call to `getCaseByDocketNumber` to account for name and title of court

File: shared/src/business/useCases/trialSessions/generateNoticeOfTrialIssuedInteractor.test.ts [hunk 5]
```diff
@@ -25,6 +25,13 @@ describe('generateNoticeOfTrialIssuedInteractor', () => {
         trialLocation: 'Boise, Idaho',
       }));
 
+    applicationContext
```
Comment (ujahio) [inline shared/src/business/useCases/trialSessions/generateNoticeOfTrialIssuedInteractor.test.ts:28]: update test to reflect calls with name and title of clerk properties

File: shared/src/business/utilities/documentGenerators/noticeOfChangeToRemoteProceeding.ts [hunk 6]
```diff
@@ -7,6 +7,13 @@ import ReactDOM from 'react-dom/server';
 export const noticeOfChangeToRemoteProceeding = async ({
   applicationContext,
   data,
+}: {
+  applicationContext: IApplicationContext;
+  data: {
+    nameOfClerk: string;
+    titleOfClerk: string;
+    [key: string]: any;
```
Comment (ujahio) [inline shared/src/business/utilities/documentGenerators/noticeOfChangeToRemoteProceeding.ts]: consider if its easy to mock out the rest of `data`

File: shared/src/business/utilities/documentGenerators/noticeOfReceiptOfPetition.ts [hunk 7]
```diff
@@ -6,6 +6,20 @@ import ReactDOM from 'react-dom/server';
 export const noticeOfReceiptOfPetition = async ({
   applicationContext,
   data,
+}: {
+  applicationContext: IApplicationContext;
+  data: {
+    accessCode?: string;
+    caseCaptionExtension: string;
+    caseTitle: string;
+    contact: any; // get proper type
```
Comment (ujahio) [inline shared/src/business/utilities/documentGenerators/noticeOfReceiptOfPetition.ts]: get proper type

File: shared/src/business/utilities/documentGenerators/noticeOfTrialIssued.ts [hunk 8]
```diff
@@ -4,7 +4,17 @@ import { generateHTMLTemplateForPDF } from '../generateHTMLTemplateForPDF/genera
...
-export const noticeOfTrialIssued = async ({ applicationContext, data }) => {
+export const noticeOfTrialIssued = async ({
+  applicationContext,
+  data,
+}: {
+  applicationContext: IApplicationContext;
+  data: {
+    nameOfClerk: string;
+    titleOfClerk: string;
+    [key: string]: any;
```
Comment (ujahio) [inline shared/src/business/utilities/documentGenerators/noticeOfTrialIssued.ts]: update type

File: shared/src/business/utilities/documentGenerators/noticeOfTrialIssuedInPerson.ts [hunk 9]
```diff
@@ -7,6 +7,13 @@ import ReactDOM from 'react-dom/server';
 export const noticeOfTrialIssuedInPerson = async ({
   applicationContext,
   data,
+}: {
+  applicationContext: IApplicationContext;
+  data: {
+    nameOfClerk: string;
+    titleOfClerk: string;
+    [key: string]: any;
```
Comment (ujahio) [inline shared/src/business/utilities/documentGenerators/noticeOfTrialIssuedInPerson.ts]: update type

File: shared/src/business/utilities/documentGenerators/order.ts [hunk 10]
```diff
@@ -4,19 +4,22 @@ import { generateHTMLTemplateForPDF } from '../generateHTMLTemplateForPDF/genera
 import React from 'react';
 import ReactDOM from 'react-dom/server';
 
+// todo: type
```
Comment (ujahio) [inline shared/src/business/utilities/documentGenerators/order.ts]: update type

File: shared/src/business/utilities/pdfGenerator/components/OrderDocketHeader.tsx [hunk 11]
```diff
@@ -6,8 +6,14 @@ export const OrderDocketHeader = ({
...
   orderTitle,
+}: {
+  caseCaptionExtension: string;
+  caseTitle: string;
+  docketNumberWithSuffix: string;
+  orderTitle?: string;
+  addedDocketNumbers?: string[];
 }) => {
-  const isEtAlOrder = addedDocketNumbers?.length > 1;
+  const isEtAlOrder = addedDocketNumbers && addedDocketNumbers?.length > 1;
```
Comment (ujahio) [inline shared/src/business/utilities/pdfGenerator/components/OrderDocketHeader.tsx]: revert code and update type

File: shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfChangeToInPersonProceeding.tsx [hunk 12]
```diff
@@ -1,13 +1,23 @@
...
+  nameOfClerk,
+  titleOfClerk,
   trialInfo,
+}: {
+  caseCaptionExtension: string;
+  caseTitle: string;
+  docketNumberWithSuffix: string;
+  nameOfClerk: string;
+  titleOfClerk: string;
+  trialInfo: FormattedTrialInfo; // confirm
```
Comment (ujahio) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfChangeToInPersonProceeding.tsx]: remove comment

File: shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfChangeToRemoteProceeding.tsx [hunk 13]
```diff
@@ -7,7 +7,16 @@ export const NoticeOfChangeToRemoteProceeding = ({
...
+  nameOfClerk,
+  titleOfClerk,
   trialInfo,
+}: {
+  nameOfClerk: string;
+  titleOfClerk: string;
+  caseCaptionExtension: string;
+  caseTitle: string;
+  docketNumberWithSuffix: string;
+  trialInfo: any;
```
Comment (ujahio) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfChangeToRemoteProceeding.tsx]: update type to `FormattedTrialInfo`

File: shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfReceiptOfPetition.tsx [hunk 14]
```diff
@@ -55,9 +55,22 @@ export const NoticeOfReceiptOfPetition = ({
...
+  nameOfClerk,
   preferredTrialCity,
   receivedAtFormatted,
   servedDate,
+  titleOfClerk,
+}: {
+  accessCode?: string;
+  caseCaptionExtension: string;
+  caseTitle: string;
+  contact: any; // get proper type
```
Comment (ujahio) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfReceiptOfPetition.tsx]: get proper type

File: shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfTrialIssued.tsx [hunk 15]
```diff
@@ -7,7 +7,13 @@ export const NoticeOfTrialIssued = ({
   caseCaptionExtension,
   caseTitle,
   docketNumberWithSuffix,
+  nameOfClerk,
+  titleOfClerk,
   trialInfo,
+}: {
+  nameOfClerk: string;
+  titleOfClerk: string;
+  [key: string]: any;
```
Comment (ujahio) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfTrialIssued.tsx]: update type

File: shared/src/business/utilities/pdfGenerator/documentTemplates/Order.tsx [hunk 16]
```diff
@@ -1,8 +1,20 @@
...
 
-export const Order = ({ options, orderContent, orderTitle, signatureText }) => {
+export const Order = ({
+  nameOfClerk,
+  options,
+  orderContent,
+  orderTitle,
+  titleOfClerk,
+}: {
+  options: any; // todo: type
```
Comment (ujahio) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/Order.tsx]: update type. Use types from `shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfDocketChange.tsx`

File: shared/src/proxies/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlProxy.ts [hunk 17]
```diff
@@ -1,25 +1,19 @@
...
-    signatureText,
+    eventCode,
+  }: {
+    addedDocketNumbers: string[];
+    contentHtml: string;
+    docketNumber: string;
+    documentTitle: string;
+    eventCode: string;
   },
 ) => {
```
Comment (ujahio) [inline shared/src/proxies/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlProxy.ts:18]: add a return type?

File: web-client/src/presenter/actions/CourtIssuedOrder/createOrderAction.test.ts [hunk 18]
```diff
@@ -23,7 +23,7 @@ describe('createOrderAction', () => {
     presenter.providers.applicationContext = applicationContextForClient;
   });
 
-  it('creates an order', async () => {
+  it('sets the content html and document title for creating orders', async () => {
```
Comment (ujahio) [inline web-client/src/presenter/actions/CourtIssuedOrder/createOrderAction.test.ts]: include eventCode in description

File: shared/src/business/useCases/trialSessions/generateNoticeOfChangeToRemoteProceedingInteractor.ts [hunk 19]
```diff
@@ -7,22 +7,13 @@ import { formatPhoneNumber } from '../../utilities/formatPhoneNumber';
...
- * @returns {Uint8Array} notice of trial session pdf
- */
 export const generateNoticeOfChangeToRemoteProceedingInteractor = async (
   applicationContext: IApplicationContext,
   {
     docketNumber,
     trialSessionInformation,
-  }: { docketNumber; trialSessionInformation: any },
-) => {
+  }: { docketNumber; trialSessionInformation: any }, // type trialSessionInformation
```
Comment (ujahio) [inline shared/src/business/useCases/trialSessions/generateNoticeOfChangeToRemoteProceedingInteractor.ts]: add type

**Review by pixiwyn — APPROVED** [review]

**Review by nechama-krigsman — APPROVED** [review]

**Review by codyseibert — APPROVED** [review]

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**